politeiawww/proposals.go

```diff
@@ -890,6 +899,126 @@ func (p *politeiawww) processProposalDetails(propDetails www.ProposalsDetails, u
 	return &reply, nil
 }
 
+// cacheVoteSumamary stores a given VoteSummary in memory.  This is to only
+// be used for proposals whose voting period has ended so that we don't have
+// to worry about cache invalidation issues.
+//
+// This function must be called without the lock held.
+func (p *politeiawww) cacheVoteSummary(token string, voteSummary www.VoteSummary) {
+	p.Lock()
+	defer p.Unlock()
+
+	p.voteSummaries[token] = voteSummary
+}
+
+// getVoteSummaries fetches the voting summary information for a set of
+// proposals.
+func (p *politeiawww) getVoteSummaries(tokens []string, bestBlock uint64) (map[string]www.VoteSummary, error) {
+
+	voteSummaries := make(map[string]www.VoteSummary)
+	tokensToLookup := make([]string, 0, len(tokens))
+
+	p.Lock()
+	for _, token := range tokens {
+		vs, ok := p.voteSummaries[token]
+		if ok {
+			voteSummaries[token] = vs
+		} else {
+			tokensToLookup = append(tokensToLookup, token)
+		}
+	}
+	p.Unlock()
+
+	if len(tokensToLookup) == 0 {
+		return voteSummaries, nil
+	}
+
+	r, err := p.decredBatchVoteSummary(tokens)
+	if err != nil {
+		return nil, err
+	}
+
+	for token, summary := range r.Summaries {
+		results := convertVoteOptionResultsFromDecred(summary.Results)
+
+		endHeight, err := strconv.ParseUint(summary.EndHeight, 10, 64)
+		if err != nil {
+			endHeight = 0
```

Comment:
We should probably do a `log.Errorf` here to alert us that something is amiss with the cache data.